tests/core/test_core_unique_ptr.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "vecmem/memory/host_memory_resource.hpp"
+#include "vecmem/memory/memory_resource.hpp"
+#include "vecmem/memory/unique_ptr.hpp"
+
+namespace {
+static vecmem::host_memory_resource g_mr;
+
+static std::size_t g_construct = 0;
+static std::size_t g_destruct = 0;
+
+class A {
+public:
+    A(void) { ++g_construct; }
+
+    ~A() { ++g_destruct; }
+};
+
+class B {
+public:
+    B(void) = delete;
+
+    B(int, float) { ++g_construct; }
+
+    ~B() { ++g_destruct; }
+};
+}  // namespace
+
+TEST(CoreUniqueObjPtrTest, EmptyPointerPrimitive) {
```

Comment:
I like `TEST_F` a little better, as it ensures that you've attached the tests to the same test case. While with `TEST` you can make typos, and the compiler doesn't complain about it.

Though this is a highly subjective comment... Even if it would also allow you to put the memory resource into the test case, along with the test classes and counters. Which would even allow (with some additional tweaks) to get rid of the `static`-ness of the counters all together.

Reply:
I've moved all the tests into fixtures and gotten rid of the global variables. I also split the tests into two files, to make it even clearer that there are two sets of tests here. Let me know if this is to your liking.